lib/galaxy/jobs/handler.py

```diff
@@ -415,11 +415,17 @@ def __filter_jobs_with_invalid_input_states(self, jobs):
             pause_message = ", ".join(jobs_to_pause[job_id])
             pause_message = "%s. To resume this job fix the input dataset(s)." % pause_message
             job, job_wrapper = self.job_pair_for_id(job_id)
-            job_wrapper.pause(job=job, message=pause_message)
+            try:
+                job_wrapper.pause(job=job, message=pause_message)
+            except Exception:
```

Comment:
So should this be ObjectNotFound ideally?

Reply:
ObjectNotFound should probably be caught at a lower level. I just wanted to make sure that *any* exception in these method don't kill the monitor loop, which effectively kills the handler until you correct whatever the problem is, possibly in the database, by hand.

Reply:
Well I'll defer to you. I don't have opinion on this. If you take this out of WIP I'll merge.